src/main/java/com/simibubi/create/api/event/ContraptionEvent.java

```diff
@@ -0,0 +1,21 @@
+package com.simibubi.create.api.event;
+
+import com.simibubi.create.content.contraptions.AbstractContraptionEntity;
+
+import net.neoforged.neoforge.event.entity.EntityEvent;
+
+public abstract class ContraptionEvent extends EntityEvent {
```

Comment:
`getEntity` should be overidden to return an AbstractContraptionEntity.
Please add javadoc too.